rust/xaynet-sdk/src/state_machine/phases/sending.rs

```diff
@@ -0,0 +1,114 @@
+use async_trait::async_trait;
+use serde::{Deserialize, Serialize};
+use tracing::{debug, error, info};
+
+use crate::{
+    state_machine::{
+        phases::Sum2,
+        IntoPhase,
+        Phase,
+        PhaseIo,
+        Progress,
+        State,
+        Step,
+        TransitionOutcome,
+        IO,
+    },
+    MessageEncoder,
+};
+
+/// Sending message phase data
+#[derive(Serialize, Deserialize, Debug)]
+pub struct Sending {
+    /// The message to send
+    message: MessageEncoder,
+
+    /// Chunk that couldn't be sent and should be tried again
+    failed: Option<Vec<u8>>,
+
+    /// State of the phase to transition to, after this one completes
+    next: Next,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+enum Next {
```

Comment:
i'd suggest to implement a similar boxing as for the coordinator phase enums, since those variants contain 264 vs 0 bytes. (i'm actually not sure why clippy didn't say sth, maybe because of some binary szt enum optimization.)

Reply:
refactored, see below.